gym-service/src/main/java/org/tolking/controller/TraineeController.java

```diff
@@ -144,7 +152,37 @@ public void addTraining(@NotNull Principal principal,
         throwExceptionIfHasError(bindingResult);
 
         Trainee trainee = traineeService.getTraineeByUsername(principal.getName());
-        trainingService.createTraining(trainee, trainingDTO);
+        TrainingEventDTO createdTraining = getADDEventDTO(trainingDTO, trainee);
+
+        //Create event
+        try {
+            trainingEventClient.createEvent(createdTraining);
+        } catch (FeignException.FeignClientException e) {
+            log.warn("Tracking service is not available");
+        }
+    }
+
+
+
+    private TrainingEventDTO getADDEventDTO(TrainingDTO trainingDTO, Trainee trainee) {
```

Comment:
```suggestion
    private TrainingEventDTO createAddEventDto(TrainingDTO trainingDTO, Trainee trainee) {
```
I don't have strong preferences between `get` or `create` here, but probably better to make verb more closer to actual process.
Also, I think it's better to extract all the logic begin eventing (including try-catch) to some service class. It could be your `TrainingService` or some new eventing service. Also, controller seem to be not good place to generate events, I'd do everything on service layer during actual training creation